packages/vite/src/node/config.ts

```diff
@@ -472,8 +473,8 @@ export async function resolveConfig(
   )
 
   const clientAlias = [
-    { find: /^\/?@vite\/env/, replacement: ENV_ENTRY },
-    { find: /^\/?@vite\/client/, replacement: CLIENT_ENTRY },
+    { find: /^\/?@vite\/env/, replacement: FS_PREFIX + ENV_ENTRY },
+    { find: /^\/?@vite\/client/, replacement: FS_PREFIX + CLIENT_ENTRY },
```

Comment:
I wonder if we should also clean up and do `path.posix.join` instead. It's done at , 
https://github.com/vitejs/vite/blob/1a8af8df9e65592fd8b23609e7fe2da3931e84ce/packages/vite/src/node/plugins/asset.ts#L253
https://github.com/vitejs/vite/blob/1a8af8df9e65592fd8b23609e7fe2da3931e84ce/packages/vite/src/node/plugins/importAnalysis.ts#L348
but not at 
https://github.com/vitejs/vite/blob/1a8af8df9e65592fd8b23609e7fe2da3931e84ce/packages/vite/src/node/server/moduleGraph.ts#L228

Reply:
Forgot to reply 😅 I don't really have a hard opinion if we want to abstract it, but usually I lean on not abstracting things until it becomes a problem.

---

Also maybe we can go ahead with this PR too now.

Reply:
I think maybe we wait for this one a bit more. If https://github.com/vitejs/vite/pull/12450 is merged, then we will win a bit, but if not, I think we won't be winning much.